ConsoleGame.java

```diff
@@ -3,9 +3,13 @@
 public class ConsoleGame implements GameType {
 
     private final Writer writer;
+    private InputStream stream;
+    private BufferedReader reader;
 
-    public ConsoleGame(Writer writer) {
+    public ConsoleGame(Writer writer, InputStream stream) {
         this.writer = writer;
+        this.stream = stream;
```

Comment:
do you need this _stream_ as a member variable?  If it isn't used other than in constructor then lose it.